src/config/backends/extlinux.ts

```diff
@@ -0,0 +1,262 @@
+import * as _ from 'lodash';
+import { fs } from 'mz';
+
+import {
+	ConfigOptions,
+	DeviceConfigBackend,
+	bootMountPoint,
+	remountAndWriteAtomic,
+} from './backend';
+import {
+	ExtlinuxFile,
+	Directive,
+	AppendDirective,
+	FDTDirective,
+} from './extlinux-file';
+import * as constants from '../../lib/constants';
+import log from '../../lib/supervisor-console';
+import { ExtLinuxParseError } from '../../lib/errors';
+
+/**
+ * A backend to handle extlinux host configuration
+ *
+ * Supports:
+ * 	- {BALENA|RESIN}_HOST_EXTLINUX_isolcpus = value | "value" | "value1","value2"
+ * 	- {BALENA|RESIN}_HOST_EXTLINUX_fdt = value | "value"
+ */
+
+export class ExtlinuxConfigBackend extends DeviceConfigBackend {
+	private static bootConfigVarPrefix = `${constants.hostConfigVarPrefix}EXTLINUX_`;
+	private static bootConfigPath = `${bootMountPoint}/extlinux/extlinux.conf`;
+	private static supportedConfigValues = ['isolcpus', 'fdt'];
+	private static supportedDirectives = ['APPEND', 'FDT'];
+
+	// These are the directives we can configure
+	private appendDirective = new AppendDirective();
+	private fdtDirective = new FDTDirective();
+
+	public static bootConfigVarRegex = new RegExp(
+		'(' + _.escapeRegExp(ExtlinuxConfigBackend.bootConfigVarPrefix) + ')(.+)',
```

Comment:
Since you only seem to care about the second group you could make the first non-capturing like
```suggestion
		'(?:' + _.escapeRegExp(ExtlinuxConfigBackend.bootConfigVarPrefix) + ')(.+)',
```
and then use `$1` in the replacement instead

Reply:
added